File: Arrays/kadanes-algorithm/print-subarays.py
def printSubArrays(nums):
    n = len(nums)
    
    for start in range(n):
        for end in range(start, n):
            for i in range(start, end+1):
                print(nums[i], end="")
            print(" ", end="")
        print()

def main():
    nums = [1,2,3,4,5]
    printSubArrays(nums)
    
main()

File: Arrays/kadanes-algorithm/print-subarays.js
const printSubArrays = (nums) => {
    const n = nums.length;

    for(let start = 0; start < n; start++) {
        for(let end = start; end < n; end++) {
            const subArrays = [];
            for(let i = start; i <= end; i++) {
                subArrays.push(nums[i]);
            }
            console.log(subArrays);
        }
    }
}

function main() {
    const nums = [1,2,3,4,5];
    printSubArrays(nums);
}

main();